### PR TITLE
Implement VehicleScheduleFrame komponenten i netex-parser-java bib.

### PR DESCRIPTION
**Pull request by gitpiphub**
Det ble opprettet en ny Java-klasse ved navn VehicleScheduleFrameParser, samt endringer i andre klasser som gjør det mulig å parse en ZIP-fil med XML-filer som inneholder VehicleScheduleFrame, blocks elementer. Det er også tilgjengelig en Test-klasse for testing.

### CODE REVIEW

File: src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    private void parseBlock(Block_VersionStructure element) {
+        if (element instanceof Block) {
+            blocks.add((Block) element);
+        } else {
+            informOnElementIntentionallySkipped(LOG, element);
+        }
+    }
+
+    private void parseTypeOfValues(TypesOfValueInFrame_RelStructure typesOfValue) {
```
Comment (testower) [inline src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java:49]: This method is not used

File: src/test/java/org/entur/netex/VehicleScheduleFrameTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        List<Block> mappedBlocks = new ArrayList<>();
+        for (Block block : blocks) {
+            List<String> dayTypes = new ArrayList<>();
+            if (block.getDayTypes() != null && block.getDayTypes().getDayTypeRef() != null) {
+                for (JAXBElement<? extends DayTypeRefStructure> dayTypeRef : block.getDayTypes().getDayTypeRef()) {
+                    dayTypes.add(dayTypeRef.getValue().getValue());
+                }
+            }
+
+            List<String> journeys = new ArrayList<>();
```
Comment (testower) [inline src/test/java/org/entur/netex/VehicleScheduleFrameTest.java:43]: What is the purpose of the dayTypes and journeys lists? They don't seem to be used in any assertions.

**Review by testower — CHANGES_REQUESTED** [review]
Have a look at my comments. I'd like to run this through @vpaturet as well who is more familiar with this part of NeTEx

File: src/test/java/org/entur/netex/VehicleScheduleFrameTest.java (hunk 2 above)
Comment (testower) [reply src/test/java/org/entur/netex/VehicleScheduleFrameTest.java:43]: The privateCode string is not used either

File: src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java (hunk 1 above)
Comment (gitpiphub) [reply src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java]: > This method is not used

Fixed

File: src/test/java/org/entur/netex/VehicleScheduleFrameTest.java (hunk 2 above)
Comment (gitpiphub) [reply src/test/java/org/entur/netex/VehicleScheduleFrameTest.java]: > What is the purpose of the mappedBlocks, dayTypes and journeys lists? They don't seem to be used in any assertions.

Fixed

File: src/test/java/org/entur/netex/VehicleScheduleFrameTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import org.entur.netex.index.api.NetexEntitiesIndex;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.rutebanken.netex.model.Block;
+import org.rutebanken.netex.model.DayTypeRefStructure;
+import org.rutebanken.netex.model.JourneyRefs_RelStructure;
+import org.rutebanken.netex.model.VehicleScheduleFrame;
+
+public class VehicleScheduleFrameTest {
```
[the remark this answers is not in the document]
Comment (gitpiphub) [reply src/test/java/org/entur/netex/VehicleScheduleFrameTest.java]: > public modifier can be removed

Fixed

File: src/main/java/org/entur/netex/index/api/NetexEntitiesIndex.java [hunk 4]
```diff
@@ -273,6 +275,8 @@ public interface NetexEntitiesIndex {
      */
     NetexEntityIndex<RoutePoint> getRoutePointIndex();
 
+    NetexEntityIndex<Block> getBlockIndex();
```
Comment (testower) [inline src/main/java/org/entur/netex/index/api/NetexEntitiesIndex.java:278]: please add a javadoc here

File: src/main/java/org/entur/netex/index/api/NetexEntitiesIndex.java [hunk 5]
```diff
@@ -292,6 +296,9 @@ public interface NetexEntitiesIndex {
     Collection<SiteFrame> getSiteFrames();
     Collection<ServiceFrame> getServiceFrames();
     Collection<ServiceCalendarFrame> getServiceCalendarFrames();
+
```
Comment (testower) [inline src/main/java/org/entur/netex/index/api/NetexEntitiesIndex.java:299]: empty lines around this is not needed

File: src/main/java/org/entur/netex/index/impl/NetexEntitiesIndexImpl.java [hunk 6]
```diff
@@ -93,6 +95,8 @@ public class NetexEntitiesIndexImpl implements NetexEntitiesIndex {
     public final VersionedNetexEntityIndex<Parking> parkingById;
     public final VersionedNetexEntityIndex<ScheduledStopPoint> scheduledStopPointById;
     public final NetexEntityIndex<RoutePoint> routePointById;
+    public final NetexEntityIndex<Block> blockById;
+
```
Comment (testower) [inline src/main/java/org/entur/netex/index/impl/NetexEntitiesIndexImpl.java:99]: remove empty line

File: src/main/java/org/entur/netex/loader/parser/NetexDocumentParser.java [hunk 7]
```diff
@@ -61,13 +62,16 @@ private void parseCommonFrame(Common_VersionFrameStructure value) {
...
+            netexIndex.getVehicleScheduleFrames().add((VehicleScheduleFrame) value);
+            parse((VehicleScheduleFrame) value, new VehicleScheduleFrameParser());
+        } else if (value instanceof TimetableFrame) {
             netexIndex.getTimetableFrames().add((TimetableFrame) value);
             parse((TimetableFrame) value, new TimeTableFrameParser());
         } else if(value instanceof ServiceFrame) {
             netexIndex.getServiceFrames().add((ServiceFrame) value);
             parse((ServiceFrame) value, new ServiceFrameParser(
-                netexIndex.getFlexibleStopPlaceIndex()
+                    netexIndex.getFlexibleStopPlaceIndex()
```
Comment (testower) [inline src/main/java/org/entur/netex/loader/parser/NetexDocumentParser.java:74]: unexpected indentation on this line and on line 90

File: src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(VehicleScheduleFrameParser.class);
+
+    private final Collection<Block> blocks = new ArrayList<>();
+
+    @Override
+    void parse(VehicleSchedule_VersionFrameStructure frame) {
+        parseBlocks(frame.getBlocks());
+    }
+
+    private void parseBlocks(BlocksInFrame_RelStructure elements) {
```
Comment (vpaturet) [inline src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java:25]: you can move private methods to the end of the class

**Review by vpaturet — CHANGES_REQUESTED** [review]

File: src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java (hunk 8 above)
Comment (gitpiphub) [reply src/main/java/org/entur/netex/loader/parser/VehicleScheduleFrameParser.java]: fixed

**Review by vpaturet — APPROVED** [review]

**Pull request merged**